### ISSUE TITLE
Refactor: Move switch host attributes/props to shadow DOM 

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-design-system#3133]
### Summary

#3091 modified `calcite-switch`'s host to preserve focus styling and to pass accessibility tests (https://github.com/Esri/calcite-components/pull/3091/commits/0a62d68d474e685585e19f5ced9d90105d003561, https://github.com/Esri/calcite-components/pull/3091/commits/bb3ce8b669e7f1e76fcad067347912d95d3c8e6e). These changes should be moved back into the shadow DOM without affecting focus styles nor tests.

cc @driskull

**Comment by driskull** [comment]
This would be nice to do pre 1.0 release.

**Comment by jcfranco** [comment]
This was addressed by https://github.com/Esri/calcite-design-system/pull/2777. Closing!
